{"uid":"27d68a775c6e76fd","name":"test_login_malformed_payload[missing_password_xfail]","fullName":"tests.auth.test_auth_api.TestAuthenticationAPI#test_login_malformed_payload","historyId":"d6083c355ddd24c2e73e7a89e0116e35","time":{"start":1744420061198,"stop":1744420061224,"duration":26},"description":"\nVerify API behavior with malformed payload.\n\nОжидаемый результат: код состояния 400 Bad Request.\n","descriptionHtml":"<p>Verify API behavior with malformed payload.</p>\n<p>Ожидаемый результат: код состояния 400 Bad Request.</p>\n","status":"skipped","statusMessage":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","statusTrace":"self = <tests.auth.test_auth_api.TestAuthenticationAPI object at 0x7fe4e4938770>\nauth_client = <api.auth.client.AuthClient object at 0x7fe4e482aa50>\npayload_dict = {'login': 'testUser4@test.com'}, expected_status = 400\ndescription = 'No password field'\n\n    @pytest.mark.negative\n    @pytest.mark.parametrize(\n        (\"payload_dict\", \"expected_status\", \"description\"),\n        [\n            pytest.param(\n                {\"login\": TEST_USER_LOGIN},\n                400,\n                \"No password field\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"missing_password_xfail\",\n            ),\n            pytest.param(\n                {\"password\": TEST_USER_PASSWORD},\n                400,\n                \"No login field\",\n                id=\"missing_login_400\",\n            ),\n            pytest.param({}, 400, \"Empty request body\", id=\"empty_payload_400\"),\n            pytest.param(\n                {\"login\": \"\", \"password\": TEST_USER_PASSWORD},\n                400,\n                \"Empty string in the login field\",\n                id=\"empty_login_str_400\",\n            ),\n            pytest.param(\n                {\"login\": TEST_USER_LOGIN, \"password\": \"\"},\n                400,\n                \"Empty string in the password field\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"empty_pass_str_xfail\",\n            ),\n            pytest.param(\n                {\"login\": \"not email\", \"password\": TEST_USER_PASSWORD},\n                400,\n                \"Incorrect login format (not email)\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"bad_login_format_xfail\",\n            ),\n        ],\n    )\n    def test_login_malformed_payload(\n        self,\n        auth_client: AuthClient,\n        payload_dict: dict[str, str | None],\n        expected_status: int,\n        description: str,\n    ) -> None:\n        \"\"\"\n        Verify API behavior with malformed payload.\n    \n        Ожидаемый результат: код состояния 400 Bad Request.\n        \"\"\"\n        logger.info(\"\\nTest: %s\", description)\n        endpoint = APIEndpoints.AUTH\n        response = auth_client.http.post(endpoint=endpoint.format(), json=payload_dict)\n>       assert response.status == expected_status, (\n            f\"Expected status {expected_status}, got {response.status}\"\n        )\nE       AssertionError: Expected status 400, got 500\nE       assert 500 == 400\nE        +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status\n\ntests/auth/test_auth_api.py:133: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"base_url","time":{"start":1744420060715,"stop":1744420060715,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"pytestconfig","time":{"start":1744420060715,"stop":1744420060715,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"_verify_url","time":{"start":1744420060715,"stop":1744420060716,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"delete_output_dir","time":{"start":1744420060715,"stop":1744420060715,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_instance","time":{"start":1744420060716,"stop":1744420060976,"duration":260},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"api_request_context","time":{"start":1744420060976,"stop":1744420060996,"duration":20},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"http_client","time":{"start":1744420060996,"stop":1744420060997,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"auth_client","time":{"start":1744420060997,"stop":1744420060997,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"\nVerify API behavior with malformed payload.\n\nОжидаемый результат: код состояния 400 Bad Request.\n","status":"skipped","statusMessage":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","statusTrace":"self = <tests.auth.test_auth_api.TestAuthenticationAPI object at 0x7fe4e4938770>\nauth_client = <api.auth.client.AuthClient object at 0x7fe4e482aa50>\npayload_dict = {'login': 'testUser4@test.com'}, expected_status = 400\ndescription = 'No password field'\n\n    @pytest.mark.negative\n    @pytest.mark.parametrize(\n        (\"payload_dict\", \"expected_status\", \"description\"),\n        [\n            pytest.param(\n                {\"login\": TEST_USER_LOGIN},\n                400,\n                \"No password field\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"missing_password_xfail\",\n            ),\n            pytest.param(\n                {\"password\": TEST_USER_PASSWORD},\n                400,\n                \"No login field\",\n                id=\"missing_login_400\",\n            ),\n            pytest.param({}, 400, \"Empty request body\", id=\"empty_payload_400\"),\n            pytest.param(\n                {\"login\": \"\", \"password\": TEST_USER_PASSWORD},\n                400,\n                \"Empty string in the login field\",\n                id=\"empty_login_str_400\",\n            ),\n            pytest.param(\n                {\"login\": TEST_USER_LOGIN, \"password\": \"\"},\n                400,\n                \"Empty string in the password field\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"empty_pass_str_xfail\",\n            ),\n            pytest.param(\n                {\"login\": \"not email\", \"password\": TEST_USER_PASSWORD},\n                400,\n                \"Incorrect login format (not email)\",\n                marks=pytest.mark.xfail(\n                    reason=\"API erratically returns 500 instead of 400\", raises=AssertionError\n                ),\n                id=\"bad_login_format_xfail\",\n            ),\n        ],\n    )\n    def test_login_malformed_payload(\n        self,\n        auth_client: AuthClient,\n        payload_dict: dict[str, str | None],\n        expected_status: int,\n        description: str,\n    ) -> None:\n        \"\"\"\n        Verify API behavior with malformed payload.\n    \n        Ожидаемый результат: код состояния 400 Bad Request.\n        \"\"\"\n        logger.info(\"\\nTest: %s\", description)\n        endpoint = APIEndpoints.AUTH\n        response = auth_client.http.post(endpoint=endpoint.format(), json=payload_dict)\n>       assert response.status == expected_status, (\n            f\"Expected status {expected_status}, got {response.status}\"\n        )\nE       AssertionError: Expected status 400, got 500\nE       assert 500 == 400\nE        +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status\n\ntests/auth/test_auth_api.py:133: AssertionError","steps":[],"attachments":[{"uid":"9c1312b13e4b8326","name":"Status Code: 500","source":"9c1312b13e4b8326.txt","type":"text/plain","size":3},{"uid":"fbd1a8ba8657fee0","name":"Request URL","source":"fbd1a8ba8657fee0.uri","type":"text/uri-list","size":30},{"uid":"17209e86aa7f6932","name":"Response Headers (JSON)","source":"17209e86aa7f6932.json","type":"application/json","size":353},{"uid":"ea625534e84849e5","name":"Response Body (Text)","source":"ea625534e84849e5.txt","type":"text/plain","size":1033},{"uid":"c8af2efb159e0327","name":"log","source":"c8af2efb159e0327.txt","type":"text/plain","size":396},{"uid":"f6b16225c409fb74","name":"stderr","source":"f6b16225c409fb74.txt","type":"text/plain","size":308}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":6,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"api_request_context::0","time":{"start":1744420061510,"stop":1744420061517,"duration":7},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_instance::0","time":{"start":1744420061518,"stop":1744420061525,"duration":7},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"negative"},{"name":"tag","value":"@pytest.mark.xfail(reason='API erratically returns 500 instead of 400', raises=<class 'AssertionError'>)"},{"name":"tag","value":"auth"},{"name":"parentSuite","value":"tests.auth"},{"name":"suite","value":"test_auth_api"},{"name":"subSuite","value":"TestAuthenticationAPI"},{"name":"host","value":"fv-az1984-790"},{"name":"thread","value":"1-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.auth.test_auth_api"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"description","value":"'No password field'"},{"name":"expected_status","value":"400"},{"name":"payload_dict","value":"{'login': 'testUser4@test.com'}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[],"history":{"statistic":{"failed":0,"broken":0,"skipped":6,"passed":0,"unknown":0,"total":6},"items":[{"uid":"5d69396d9706692","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/24//#testresult/5d69396d9706692","status":"skipped","statusDetails":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","time":{"start":1744339367640,"stop":1744339367665,"duration":25}},{"uid":"acb57a20ba957877","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/22//#testresult/acb57a20ba957877","status":"skipped","statusDetails":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","time":{"start":1744244023235,"stop":1744244023260,"duration":25}},{"uid":"4e4312741306c317","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/21//#testresult/4e4312741306c317","status":"skipped","statusDetails":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","time":{"start":1744243579782,"stop":1744243579807,"duration":25}},{"uid":"9e62b1f952123dde","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/20//#testresult/9e62b1f952123dde","status":"skipped","statusDetails":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","time":{"start":1744243348381,"stop":1744243348408,"duration":27}},{"uid":"f94aa01fb8cc3231","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/19//#testresult/f94aa01fb8cc3231","status":"skipped","statusDetails":"XFAIL API erratically returns 500 instead of 400\n\nAssertionError: Expected status 400, got 500\nassert 500 == 400\n +  where 500 = <APIResponse url='http://localhost:4040/api/auth' status=500 status_text='Internal Server Error'>.status","time":{"start":1744242336458,"stop":1744242336483,"duration":25}}]},"tags":["@pytest.mark.xfail(reason='API erratically returns 500 instead of 400', raises=<class 'AssertionError'>)","negative","auth"]},"source":"27d68a775c6e76fd.json","parameterValues":["'No password field'","400","{'login': 'testUser4@test.com'}"]}